lib/bap_types/bap_eval.ml

```diff
@@ -88,7 +93,7 @@ module Make2(State : Monad.S2) = struct
       self#eval exp >>= self#storage_of_value
 
     method eval_load ~mem ~addr endian sz =
-      if Size.(sz <> `r8)
+      if Size.(sz <> infer_value_size mem)
```

Comment:
`infer_value_size` shall not raise, instead `self#type_error TE.bad_mem` should be called if `mem` is not of type `Mem`.